commons/ava_testsuite/staking_gecko_network_tests.go

```diff
@@ -92,26 +148,29 @@ func (test FiveNodeStakingNetworkFullyConnectedTest) Run(network interface{}, co
 
 	logrus.Debugf("Network ID Set: %+v", nodeIdSet)
 
-	// verify peer lists have set of IDs in network, except their own
+	// verify bootstrapper peer lists have full set of IDs in network.
 	for _, serviceId := range allServiceIds {
 		client, err := castedNetwork.GetGeckoClient(serviceId)
 		if err != nil {
 			context.Fatal(stacktrace.Propagate(err, "Could not get client for service with ID %v", serviceId))
 		}
 		peers, err := client.AdminApi().GetPeers()
+		nodeId, err := client.AdminApi().GetNodeId()
 		if err != nil {
 			context.Fatal(stacktrace.Propagate(err, "Could not get peers of service with ID %v", serviceId))
 		}
-		logrus.Debugf("Peer set: %+v", peers)
+		logrus.Debugf("Length of Peer set for node %v: %v", serviceId, len(peers))
 
 		peerSet := map[string]bool{}
 		for _, peer := range peers {
 			peerSet[peer.Id] = true
-			// verify that peer is inside the nodeIdSet
-			context.AssertTrue(nodeIdSet[peer.Id])
 		}
-		// verify that every other peer (besides the node itself) is represented in the peer list.
-		context.AssertTrue(len(peerSet) == len(allServiceIds) - 1)
+		for expectedNodeId, _ := range nodeIdSet {
+			// Nodes are expected to have all other nodes in the network on their peer list.
+			if nodeId != expectedNodeId {
+				context.AssertTrue(peerSet[expectedNodeId])
```

Comment:
I think this will break if the guy isn't in the set, right? Do we need to check for explicit existence?

Reply:
If which guy isn't in the set? "expectedNodeId" represents the set of nodeIds that we want to have in the peerset

Reply:
and if expectedNodeId is not in peerset, it correctly fails with an Assert Failure

Reply:
let me verify this..

Reply:
yeah, "An attempt to fetch a map value with a key that is not present in the map will return the zero value for the type of the entries in the map. " https://golang.org/doc/effective_go.html#maps so for bool maps it returns false